packages/service-library/src/servicelib/utils_debug_asyncio.py

```diff
@@ -0,0 +1,10 @@
+import asyncio
+from io import StringIO
+
+
+def get_loop_info() -> str:
+    stream = StringIO()
+    for n, task in enumerate(asyncio.all_tasks()):
+        prefix = "*" if task == asyncio.current_task() else " "
+        print(f"{prefix}{n+1:2d}) {task}", file=stream)
```

Comment:
Should this write to a logger?

Reply:
It is just a small helper function I put aside because it was very handy while debugging. I would use it as
```python
logger.debug("Bla, bla. detail: %s", get_loop_info())
```
now, i would like ``get_loop_info``  to be lazy ... any suggestions? Kind of

```python
logger.debug("Bla, bla. detail: %s", logger.log_level == logging.DEBUG and get_loop_info())
```

Reply:
I have read that logger will call ``str`` or the arguments and there are some suggestions as creating wrappers like

```python
class WrapFun:
    def __init__(self, fun):
         self._fun = fun
   def __str__(self):
        return self._fun()

logger.debug("Bla, bla. detail: %s", WrapFun(get_loop_info))
```

I guess we could also do put that inside
```python

class WrapFun:
    def __init__(self, fun):
         self._fun = fun
   def __str__(self):
        return self._fun()
   def __call__(self):
       return self._fun()

def get_loop_info():
  def _impl():
      # ...
  return WrapFun(_impl)
```

Reply:
:-(